### PR TITLE
.*: replace black-white-list by block-allow-list

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
 replace `black-white-list` by `block-allow-list`

### What is changed and how it works?

 replace `black-white-list` by `block-allow-list`, and also support `black-white-list` for compatibility

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to update the `dm/dm-ansible`
 - Need to be included in the release note

### CODE REVIEW

File: dm/portal/frontend/src/types.ts [hunk 1]
```diff
@@ -137,19 +137,19 @@ export interface IFilters {
   [key: string]: IFilter
 }
 
-// BWList: Black White List
-export interface IBWTable {
+// BAList: Block Allow List
```
Comment (WangXiangUSTC) [inline dm/portal/frontend/src/types.ts:140]: @baurine I update some frontend code, please check it

File: dm/portal/frontend/src/types.ts [hunk 2]
```diff
@@ -170,5 +170,5 @@ export interface IFinalConfig {
 
   routes: IRoutes
   filters: IFilters
-  'black-white-list': IBWList
+  'black-white-list': IBAList
```
Comment (baurine) [inline dm/portal/frontend/src/types.ts:173]: do we need to change `black-white-list`?

File: dm/portal/frontend/src/utils/config-util.ts [hunk 3]
```diff
@@ -338,7 +338,7 @@ export function genFinalConfig(
 ) {
   const routes: IRoutes = genRoutesConfig(targetSchemas, allTables)
   const filters: IFilters = genFiltersConfig(sourceSchemas, allTables)
-  const bwList: IBWList = genBlackWhiteList(allTables, autoSycnUpstream)
+  const baList: IBAList = genBlackWhiteList(allTables, autoSycnUpstream)
```
Comment (baurine) [inline dm/portal/frontend/src/utils/config-util.ts:341]: rename `genBlackWhiteList()` method?

File: dm/portal/frontend/src/utils/config-util.ts [hunk 4]
```diff
@@ -356,7 +356,7 @@ export function genFinalConfig(
 
     routes,
     filters,
-    'black-white-list': bwList
+    'black-white-list': baList
```
Comment (baurine) [inline dm/portal/frontend/src/utils/config-util.ts:359]: do we need to change `black-white-list`?

File: dm/portal/frontend/src/utils/config-util.ts [hunk 5]
```diff
@@ -475,18 +475,18 @@ export function parseFinalConfig(finalConfig: IFinalConfig) {
   const allTables: IFullTables = {}
 
   // 第一步：从 black-white-lists 中还原得到 sourceFullInstances, sourceSchemas, allTables
-  const bwList: IBWList = finalConfig['black-white-list']
-  Object.keys(bwList).forEach(bwListKey => {
-    // bwListKey => "replica-1.bw_list.1"
-    const sourceId: string = bwListKey.split('.')[0]
+  const baList: IBAList = finalConfig['black-white-list']
```
Comment (baurine) [inline dm/portal/frontend/src/utils/config-util.ts:478]: mark

File: dm/portal/frontend/src/utils/config-util.ts [hunk 6]
```diff
@@ -475,18 +475,18 @@ export function parseFinalConfig(finalConfig: IFinalConfig) {
   const allTables: IFullTables = {}
 
   // 第一步：从 black-white-lists 中还原得到 sourceFullInstances, sourceSchemas, allTables
```
Comment (baurine) [inline dm/portal/frontend/src/utils/config-util.ts:477]: `从 black-white-lists` mark

**Comment by WangXiangUSTC** [comment]
@baurine all comments addressed in https://github.com/pingcap/dm/pull/756/commits/f686cb13af233b73991009ac2dc3cd78fb05dfa6 PTAL again

**Comment by baurine** [comment]
> @baurine all comments addressed in [f686cb1](https://github.com/pingcap/dm/commit/f686cb13af233b73991009ac2dc3cd78fb05dfa6) PTAL again

LGTM for the frontend part.

File: tests/all_mode/run.sh [hunk 7]
```diff
@@ -137,6 +137,10 @@ function run() {
     # check_metric $WORKER1_PORT 'dm_syncer_replication_lag{task="test"}' 3 0 2
     # check_metric $WORKER2_PORT 'dm_syncer_replication_lag{task="test"}' 3 0 2
 
+    # test block-allow-list by the way
+    run_sql "show databases;" $TIDB_PORT $TIDB_PASSWORD
+    check_not_contains "ignore_db"
```
[the remark this answers is not in the document]
Comment (WangXiangUSTC) [reply tests/all_mode/run.sh:142]: add in 171dbdc

**Comment by WangXiangUSTC** [comment]
> How about updating tidb-tools to a replaced version (including [pingcap/tidb-tools#353](https://github.com/pingcap/tidb-tools/pull/353))?

I update the version in e2cd417

**Pull request merged**